### ISSUE TITLE
NoMethodError: undefined method `name' for nil:NilClass

### ISSUE THREAD

**Issue by bennettp123** [bundler/bundler#5982]
# Error Report

## Questions

Please fill out answers to these questions, it'll help us figure out
why things are going wrong.

- **What did you do?**

  I ran the command `/usr/local/bin/bundle install`

- **What did you expect to happen?**

  I expected Bundler to install the missing gems

- **What happened instead?**

  Instead, what happened was it produced the error below

- **Have you tried any solutions posted on similar issues in our issue tracker, stack overflow, or google?**

  Adding a Gemfile.lock seems to fix (or work around) the issue.
  I also checked for other issues, but couldn't find a report for this combination of ruby and bundler.

- **Have you read our issues document, https://github.com/bundler/bundler/blob/master/doc/contributing/ISSUES.md?**

  Yes

## Backtrace

```
NoMethodError: undefined method `name' for nil:NilClass
  /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/resolver.rb:215:in `block in start'
  /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/resolver.rb:215:in `reject'
  /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/resolver.rb:215:in `start'
  /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/resolver.rb:191:in `resolve'
  /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/definition.rb:235:in `resolve'
  /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/definition.rb:159:in `specs'
  /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/definition.rb:147:in `resolve_remotely!'
  /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/installer.rb:226:in `resolve_if_need'
  /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/installer.rb:78:in `run'
  /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/installer.rb:24:in `install'
  /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/cli/install.rb:68:in `run'
  /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/cli.rb:188:in `block in install'
  /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/settings.rb:92:in `temporary'
  /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/cli.rb:187:in `install'
  /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
  /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
  /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'
  /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/cli.rb:22:in `dispatch'
  /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/vendor/thor/lib/thor/base.rb:466:in `start'
  /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/cli.rb:13:in `start'
  /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/exe/bundle:30:in `block in <top (required)>'
  /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/lib/bundler/friendly_errors.rb:121:in `with_friendly_errors'
  /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/exe/bundle:22:in `<top (required)>'
  /usr/local/bin/bundle:23:in `load'
  /usr/local/bin/bundle:23:in `<main>'
```

## Environment

```
Bundler   1.15.4
Rubygems  2.6.13
Ruby      2.3.4p301 (2017-03-30 revision 58214) [x86_64-linux]
GEM_HOME  /usr/local/bundle
GEM_PATH  
Git       2.1.4
Platform  x86_64-linux
OpenSSL   OpenSSL 1.0.1t  3 May 2016
```

## Bundler settings

```
disable_shared_gems
  Set for your local app (/usr/local/bundle/config): true
silence_root_warning
  Set via BUNDLE_SILENCE_ROOT_WARNING: true
app_config
  Set via BUNDLE_APP_CONFIG: "/usr/local/bundle"
bin
  Set via BUNDLE_BIN: "/usr/local/bundle/bin"
path
  Set via BUNDLE_PATH: "/usr/local/bundle"
```

### Gemfile

```ruby
source ENV['GEM_SOURCE'] || 'https://rubygems.org'

def location_for(place_or_version, fake_version = nil)
  if place_or_version =~ %r{\A(git[:@][^#]*)#(.*)}
    [fake_version, { git: Regexp.last_match(1), branch: Regexp.last_match(2), require: false }].compact
  elsif place_or_version =~ %r{\Afile:\/\/(.*)}
    ['>= 0', { path: File.expand_path(Regexp.last_match(1)), require: false }]
  else
    [place_or_version, { require: false }]
  end
end

def gem_type(place_or_version)
  if place_or_version =~ %r{\Agit[:@]}
    :git
  elsif !place_or_version.nil? && place_or_version.start_with?('file:')
    :file
  else
    :gem
  end
end

ruby_version_segments = Gem::Version.new(RUBY_VERSION.dup).segments
minor_version = ruby_version_segments[0..1].join('.')

group :development do
  gem "fast_gettext", '1.1.0',                         require: false if Gem::Version.new(RUBY_VERSION.dup) < Gem::Version.new('2.1.0')
  gem "fast_gettext",                                  require: false if Gem::Version.new(RUBY_VERSION.dup) >= Gem::Version.new('2.1.0')
  gem "json_pure", '<= 2.0.1',                         require: false if Gem::Version.new(RUBY_VERSION.dup) < Gem::Version.new('2.0.0')
  gem "json", '= 1.8.1',                               require: false if Gem::Version.new(RUBY_VERSION.dup) == Gem::Version.new('2.1.9')
  gem "puppet-module-posix-default-r#{minor_version}", require: false, platforms: [:ruby]
  gem "puppet-module-posix-dev-r#{minor_version}",     require: false, platforms: [:ruby]
  gem "puppet-module-win-default-r#{minor_version}",   require: false, platforms: [:mswin, :mingw, :x64_mingw]
  gem "puppet-module-win-dev-r#{minor_version}",       require: false, platforms: [:mswin, :mingw, :x64_mingw]
end

puppet_version = ENV['PUPPET_GEM_VERSION']
puppet_type = gem_type(puppet_version)
facter_version = ENV['FACTER_GEM_VERSION']
hiera_version = ENV['HIERA_GEM_VERSION']

def puppet_older_than?(version)
  puppet_version = ENV['PUPPET_GEM_VERSION']
  !puppet_version.nil? &&
    Gem::Version.correct?(puppet_version) &&
    Gem::Requirement.new("< #{version}").satisfied_by?(Gem::Version.new(puppet_version.dup))
end

gems = {}

gems['puppet'] = location_for(puppet_version)

# If facter or hiera versions have been specified via the environment
# variables, use those versions. If not, and if the puppet version is < 3.5.0,
# use known good versions of both for puppet < 3.5.0.
if facter_version
  gems['facter'] = location_for(facter_version)
elsif puppet_type == :gem && puppet_older_than?('3.5.0')
  gems['facter'] = ['>= 1.6.11', '<= 1.7.5', require: false]
end

if hiera_version
  gems['hiera'] = location_for(ENV['HIERA_GEM_VERSION'])
elsif puppet_type == :gem && puppet_older_than?('3.5.0')
  gem['hiera'] = ['>= 1.0.0', '<= 1.3.0', require: false]
end

if Gem.win_platform? && (puppet_type != :gem || puppet_older_than?('3.5.0'))
  # For Puppet gems < 3.5.0 (tested as far back as 3.0.0) on Windows
  if puppet_type == :gem
    gems['ffi'] =            ['1.9.0',                require: false]
    gems['minitar'] =        ['0.5.4',                require: false]
    gems['win32-eventlog'] = ['0.5.3',    '<= 0.6.5', require: false]
    gems['win32-process'] =  ['0.6.5',    '<= 0.7.5', require: false]
    gems['win32-security'] = ['~> 0.1.2', '<= 0.2.5', require: false]
    gems['win32-service'] =  ['0.7.2',    '<= 0.8.8', require: false]
  else
    gems['ffi'] =            ['~> 1.9.0',             require: false]
    gems['minitar'] =        ['~> 0.5.4',             require: false]
    gems['win32-eventlog'] = ['~> 0.5',   '<= 0.6.5', require: false]
    gems['win32-process'] =  ['~> 0.6',   '<= 0.7.5', require: false]
    gems['win32-security'] = ['~> 0.1',   '<= 0.2.5', require: false]
    gems['win32-service'] =  ['~> 0.7',   '<= 0.8.8', require: false]
  end

  gems['win32-dir'] = ['~> 0.3', '<= 0.4.9', require: false]

  if RUBY_VERSION.start_with?('1.')
    gems['win32console'] = ['1.3.2', require: false]
    # sys-admin was removed in Puppet 3.7.0 and doesn't compile under Ruby 2.x
    gems['sys-admin'] =    ['1.5.6', require: false]
  end

  # Puppet < 3.7.0 requires these.
  # Puppet >= 3.5.0 gem includes these as requirements.
  # The following versions are tested to work with 3.0.0 <= puppet < 3.7.0.
  gems['win32-api'] =           ['1.4.8', require: false]
  gems['win32-taskscheduler'] = ['0.2.2', require: false]
  gems['windows-api'] =         ['0.4.3', require: false]
  gems['windows-pr'] =          ['1.2.3', require: false]
elsif Gem.win_platform?
  # If we're using a Puppet gem on Windows which handles its own win32-xxx gem
  # dependencies (>= 3.5.0), set the maximum versions (see PUP-6445).
  gems['win32-dir'] =      ['<= 0.4.9', require: false]
  gems['win32-eventlog'] = ['<= 0.6.5', require: false]
  gems['win32-process'] =  ['<= 0.7.5', require: false]
  gems['win32-security'] = ['<= 0.2.5', require: false]
  gems['win32-service'] =  ['<= 0.8.8', require: false]
end

gems.each do |gem_name, gem_params|
  gem gem_name, *gem_params
end

# Evaluate Gemfile.local and ~/.gemfile if they exist
extra_gemfiles = [
  "#{__FILE__}.local",
  File.join(Dir.home, '.gemfile'),
]

extra_gemfiles.each do |gemfile|
  if File.file?(gemfile) && File.readable?(gemfile)
    eval(File.read(gemfile), binding)
  end
end

# Added at 2017-08-27 23:44:38 +0000 by bennett:
gem "beaker", "~> 3.23"

# Added at 2017-08-27 23:48:42 +0000 by bennett:
gem "beaker-rspec", "~> 6.1"

# Added at 2017-08-28 00:26:31 +0000 by bennett:
gem "docker-api", "~> 1.33"

# Added at 2017-08-28 09:13:12 +0800 by bennett:
gem "bcrypt_pbkdf", "~> 1.0"

# Added at 2017-08-28 09:13:41 +0800 by bennett:
gem "rbnacl", "~> 4.0.2"

# Added at 2017-08-28 09:14:47 +0800 by bennett:
gem "rbnacl-libsodium", "~> 1.0"

# Added at 2017-08-28 09:19:54 +0800 by bennett:
gem "beaker-docker", "~> 0.2.0"

# Added at 2017-08-28 11:00:30 +0800 by bennett:
gem "beaker-puppet_install_helper", "~> 0.7.1"

# Added at 2017-08-28 17:22:21 +0800 by bennett:
gem "beaker-module_install_helper", "~> 0.1.5"

# vim: syntax=ruby
```

### Gemfile.lock

```
<No /pwd/Gemfile.lock found>
```

**Comment by bennettp123** [comment]
It works fine when Gemfile.lock is present (copied from a different system):

```
GEM
  remote: https://rubygems.org/
  specs:
    CFPropertyList (2.3.5)
    addressable (2.5.1)
      public_suffix (~> 2.0, >= 2.0.2)
    ansi (1.5.0)
    ast (2.3.0)
    aws-sdk-v1 (1.67.0)
      json (~> 1.4)
      nokogiri (~> 1)
    bcrypt_pbkdf (1.0.0)
    beaker (3.23.0)
      beaker-aws (~> 0.1)
      beaker-docker (~> 0.1)
      beaker-google (~> 0.1)
      beaker-hiera (~> 0.0)
      beaker-hostgenerator
      beaker-openstack (~> 0.1)
      beaker-puppet (~> 0.0)
      beaker-vagrant (~> 0.1)
      beaker-vcloud (~> 0.1)
      beaker-vmpooler (~> 1.0)
      beaker-vmware (~> 0.1)
      hocon (~> 1.0)
      in-parallel (~> 0.1)
      inifile (~> 3.0)
      minitar (~> 0.6)
      minitest (~> 5.4)
      net-scp (~> 1.2)
      net-ssh (~> 4.0)
      open_uri_redirections (~> 0.2.1)
      pry-byebug (~> 3.4.2)
      rb-readline (~> 0.5.3)
      rsync (~> 1.0.9)
      stringify-hash (~> 0.0)
      thor (~> 0.19)
    beaker-aws (0.3.0)
      aws-sdk-v1 (~> 1.57)
      stringify-hash (~> 0.0.0)
    beaker-docker (0.2.0)
      docker-api
      stringify-hash (~> 0.0.0)
    beaker-google (0.1.0)
      google-api-client (~> 0.9)
      stringify-hash (~> 0.0.0)
    beaker-hiera (0.1.1)
      stringify-hash (~> 0.0.0)
    beaker-hostgenerator (1.0.2)
      deep_merge (~> 1.0)
      stringify-hash (~> 0.0.0)
    beaker-module_install_helper (0.1.5)
      beaker (>= 2.0)
    beaker-openstack (0.1.0)
      fog (~> 1.38)
      stringify-hash (~> 0.0.0)
    beaker-puppet (0.6.0)
      in-parallel (~> 0.1)
      oga
      stringify-hash (~> 0.0.0)
    beaker-puppet_install_helper (0.7.1)
      beaker (>= 2.0)
    beaker-rspec (6.1.0)
      beaker (~> 3.0)
      rspec (~> 3.0)
      serverspec (~> 2)
      specinfra (~> 2)
    beaker-vagrant (0.1.0)
      stringify-hash (~> 0.0.0)
    beaker-vcloud (0.1.0)
      beaker-vmpooler
      rbvmomi (~> 1.9)
      stringify-hash (~> 0.0.0)
    beaker-vmpooler (1.1.0)
      stringify-hash (~> 0.0.0)
    beaker-vmware (0.2.0)
      fission (~> 0.4)
      rbvmomi (~> 1.9)
      stringify-hash (~> 0.0.0)
    builder (3.2.3)
    byebug (9.0.6)
    coderay (1.1.1)
    declarative (0.0.9)
    declarative-option (0.1.0)
    deep_merge (1.1.1)
    diff-lcs (1.3)
    docile (1.1.5)
    docker-api (1.33.6)
      excon (>= 0.38.0)
      json
    excon (0.58.0)
    facter (2.5.0)
    facterdb (0.3.12)
      facter
      jgrep
    faraday (0.13.1)
      multipart-post (>= 1.2, < 3)
    fast_gettext (1.1.0)
    ffi (1.9.18)
    fission (0.5.0)
      CFPropertyList (~> 2.2)
    fog (1.41.0)
      fog-aliyun (>= 0.1.0)
      fog-atmos
      fog-aws (>= 0.6.0)
      fog-brightbox (~> 0.4)
      fog-cloudatcost (~> 0.1.0)
      fog-core (~> 1.45)
      fog-digitalocean (>= 0.3.0)
      fog-dnsimple (~> 1.0)
      fog-dynect (~> 0.0.2)
      fog-ecloud (~> 0.1)
      fog-google (<= 0.1.0)
      fog-internet-archive
      fog-joyent
      fog-json
      fog-local
      fog-openstack
      fog-powerdns (>= 0.1.1)
      fog-profitbricks
      fog-rackspace
      fog-radosgw (>= 0.0.2)
      fog-riakcs
      fog-sakuracloud (>= 0.0.4)
      fog-serverlove
      fog-softlayer
      fog-storm_on_demand
      fog-terremark
      fog-vmfusion
      fog-voxel
      fog-vsphere (>= 0.4.0)
      fog-xenserver
      fog-xml (~> 0.1.1)
      ipaddress (~> 0.5)
      json (>= 1.8, < 2.0)
    fog-aliyun (0.2.0)
      fog-core (~> 1.27)
      fog-json (~> 1.0)
      ipaddress (~> 0.8)
      xml-simple (~> 1.1)
    fog-atmos (0.1.0)
      fog-core
      fog-xml
    fog-aws (1.4.1)
      fog-core (~> 1.38)
      fog-json (~> 1.0)
      fog-xml (~> 0.1)
      ipaddress (~> 0.8)
    fog-brightbox (0.13.0)
      fog-core (~> 1.22)
      fog-json
      inflecto (~> 0.0.2)
    fog-cloudatcost (0.1.2)
      fog-core (~> 1.36)
      fog-json (~> 1.0)
      fog-xml (~> 0.1)
      ipaddress (~> 0.8)
    fog-core (1.45.0)
      builder
      excon (~> 0.58)
      formatador (~> 0.2)
    fog-digitalocean (0.3.0)
      fog-core (~> 1.42)
      fog-json (>= 1.0)
      fog-xml (>= 0.1)
      ipaddress (>= 0.5)
    fog-dnsimple (1.0.0)
      fog-core (~> 1.38)
      fog-json (~> 1.0)
    fog-dynect (0.0.3)
      fog-core
      fog-json
      fog-xml
    fog-ecloud (0.3.0)
      fog-core
      fog-xml
    fog-google (0.1.0)
      fog-core
      fog-json
      fog-xml
    fog-internet-archive (0.0.1)
      fog-core
      fog-json
      fog-xml
    fog-joyent (0.0.1)
      fog-core (~> 1.42)
      fog-json (>= 1.0)
    fog-json (1.0.2)
      fog-core (~> 1.0)
      multi_json (~> 1.10)
    fog-local (0.3.1)
      fog-core (~> 1.27)
    fog-openstack (0.1.21)
      fog-core (>= 1.40)
      fog-json (>= 1.0)
      ipaddress (>= 0.8)
    fog-powerdns (0.1.1)
      fog-core (~> 1.27)
      fog-json (~> 1.0)
      fog-xml (~> 0.1)
    fog-profitbricks (4.0.0)
      fog-core (~> 1.42)
      fog-json (~> 1.0)
    fog-rackspace (0.1.5)
      fog-core (>= 1.35)
      fog-json (>= 1.0)
      fog-xml (>= 0.1)
      ipaddress (>= 0.8)
    fog-radosgw (0.0.5)
      fog-core (>= 1.21.0)
      fog-json
      fog-xml (>= 0.0.1)
    fog-riakcs (0.1.0)
      fog-core
      fog-json
      fog-xml
    fog-sakuracloud (1.7.5)
      fog-core
      fog-json
    fog-serverlove (0.1.2)
      fog-core
      fog-json
    fog-softlayer (1.1.4)
      fog-core
      fog-json
    fog-storm_on_demand (0.1.1)
      fog-core
      fog-json
    fog-terremark (0.1.0)
      fog-core
      fog-xml
    fog-vmfusion (0.1.0)
      fission
      fog-core
    fog-voxel (0.1.0)
      fog-core
      fog-xml
    fog-vsphere (1.12.0)
      fog-core
      rbvmomi (~> 1.9)
    fog-xenserver (0.3.0)
      fog-core
      fog-xml
    fog-xml (0.1.3)
      fog-core
      nokogiri (>= 1.5.11, < 2.0.0)
    formatador (0.2.5)
    gettext (3.2.4)
      locale (>= 2.0.5)
      text (>= 1.3.0)
    gettext-setup (0.26)
      fast_gettext (~> 1.1.0)
      gettext (>= 3.0.2)
      locale
    google-api-client (0.13.1)
      addressable (~> 2.5, >= 2.5.1)
      googleauth (~> 0.5)
      httpclient (>= 2.8.1, < 3.0)
      mime-types (~> 3.0)
      representable (~> 3.0)
      retriable (>= 2.0, < 4.0)
    googleauth (0.5.3)
      faraday (~> 0.12)
      jwt (~> 1.4)
      logging (~> 2.0)
      memoist (~> 0.12)
      multi_json (~> 1.11)
      os (~> 0.9)
      signet (~> 0.7)
    hiera (3.4.0)
    hocon (1.2.5)
    httpclient (2.8.3)
    in-parallel (0.1.17)
    inflecto (0.0.2)
    inifile (3.0.0)
    ipaddress (0.8.3)
    jgrep (1.5.0)
    json (1.8.3)
    json-schema (2.8.0)
      addressable (>= 2.4)
    jwt (1.5.6)
    little-plugger (1.1.4)
    locale (2.1.2)
    logging (2.2.2)
      little-plugger (~> 1.1)
      multi_json (~> 1.10)
    mcollective-client (2.11.2)
      json
      stomp
      systemu
    memoist (0.16.0)
    metaclass (0.0.4)
    metadata-json-lint (2.0.2)
      json-schema (~> 2.8)
      spdx-licenses (~> 1.0)
    method_source (0.8.2)
    mime-types (3.1)
      mime-types-data (~> 3.2015)
    mime-types-data (3.2016.0521)
    mini_portile2 (2.2.0)
    minitar (0.6.1)
    minitest (5.10.3)
    mocha (1.1.0)
      metaclass (~> 0.0.1)
    multi_json (1.12.1)
    multipart-post (2.0.0)
    net-scp (1.2.1)
      net-ssh (>= 2.6.5)
    net-ssh (4.1.0)
    net-telnet (0.1.1)
    nokogiri (1.8.0)
      mini_portile2 (~> 2.2.0)
    oga (2.10)
      ast
      ruby-ll (~> 2.1)
    open_uri_redirections (0.2.1)
    os (0.9.6)
    parallel (1.12.0)
    parallel_tests (2.14.2)
      parallel
    parser (2.4.0.0)
      ast (~> 2.2)
    powerpack (0.1.1)
    pry (0.10.4)
      coderay (~> 1.1.0)
      method_source (~> 0.8.1)
      slop (~> 3.4)
    pry-byebug (3.4.3)
      byebug (>= 9.0, < 9.1)
      pry (~> 0.10)
    public_suffix (2.0.5)
    puppet (5.0.1)
      facter (> 2.0, < 4)
      gettext-setup (>= 0.10, < 1)
      hiera (>= 3.2.1, < 4)
      locale (~> 2.1)
    puppet-lint (2.3.0)
    puppet-module-posix-default-r2.3 (0.1.1)
    puppet-module-posix-dev-r2.3 (0.1.1)
      gettext-setup (~> 0.26)
      metadata-json-lint (>= 2.0.2, < 3.0.0)
      mocha (>= 1.0.0, < 1.2.0)
      parallel_tests (>= 2.14.1, < 2.14.3)
      pry (~> 0.10.4)
      puppet-lint (>= 2.3.0, < 3.0.0)
      puppet-syntax (>= 2.4.1, < 3.0.0)
      puppetlabs_spec_helper (>= 2.3.1, < 3.0.0)
      rainbow (>= 2.0.0, < 2.2.0)
      rspec-puppet (>= 2.3.2, < 3.0.0)
      rspec-puppet-facts (~> 1.8)
      rspec_junit_formatter (~> 0.2)
      rubocop (~> 0.49)
      rubocop-rspec (~> 1.15)
      simplecov (>= 0.14.1, < 1.0.0)
      specinfra (= 2.67.3)
    puppet-syntax (2.4.1)
      rake
    puppetlabs_spec_helper (2.3.2)
      mocha (~> 1.0)
      puppet-lint (~> 2.0)
      puppet-syntax (~> 2.0)
      rspec-puppet (~> 2.0)
    rainbow (2.1.0)
    rake (12.0.0)
    rb-readline (0.5.5)
    rbnacl (4.0.2)
      ffi
    rbnacl-libsodium (1.0.13)
      rbnacl (>= 3.0.1)
    rbvmomi (1.11.3)
      builder (~> 3.0)
      json (>= 1.8)
      nokogiri (~> 1.5)
      trollop (~> 2.1)
    representable (3.0.4)
      declarative (< 0.1.0)
      declarative-option (< 0.2.0)
      uber (< 0.2.0)
    retriable (3.1.1)
    rspec (3.6.0)
      rspec-core (~> 3.6.0)
      rspec-expectations (~> 3.6.0)
      rspec-mocks (~> 3.6.0)
    rspec-core (3.6.0)
      rspec-support (~> 3.6.0)
    rspec-expectations (3.6.0)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.6.0)
    rspec-its (1.2.0)
      rspec-core (>= 3.0.0)
      rspec-expectations (>= 3.0.0)
    rspec-mocks (3.6.0)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.6.0)
    rspec-puppet (2.6.8)
      rspec
    rspec-puppet-facts (1.8.0)
      facter
      facterdb (>= 0.3.0)
      json
      mcollective-client
      puppet
    rspec-support (3.6.0)
    rspec_junit_formatter (0.3.0)
      rspec-core (>= 2, < 4, != 2.12.0)
    rsync (1.0.9)
    rubocop (0.49.1)
      parallel (~> 1.10)
      parser (>= 2.3.3.1, < 3.0)
      powerpack (~> 0.1)
      rainbow (>= 1.99.1, < 3.0)
      ruby-progressbar (~> 1.7)
      unicode-display_width (~> 1.0, >= 1.0.1)
    rubocop-rspec (1.15.1)
      rubocop (>= 0.42.0)
    ruby-ll (2.1.2)
      ansi
      ast
    ruby-progressbar (1.8.1)
    serverspec (2.39.0)
      multi_json
      rspec (~> 3.0)
      rspec-its
      specinfra (~> 2.53)
    sfl (2.3)
    signet (0.7.3)
      addressable (~> 2.3)
      faraday (~> 0.9)
      jwt (~> 1.5)
      multi_json (~> 1.10)
    simplecov (0.15.0)
      docile (~> 1.1.0)
      json (>= 1.8, < 3)
      simplecov-html (~> 0.10.0)
    simplecov-html (0.10.2)
    slop (3.6.0)
    spdx-licenses (1.1.0)
    specinfra (2.67.3)
      net-scp
      net-ssh (>= 2.7, < 5.0)
      net-telnet
      sfl
    stomp (1.4.4)
    stringify-hash (0.0.2)
    systemu (2.6.5)
    text (1.3.1)
    thor (0.20.0)
    trollop (2.1.2)
    uber (0.1.0)
    unicode-display_width (1.3.0)
    xml-simple (1.1.5)

PLATFORMS
  ruby

DEPENDENCIES
  bcrypt_pbkdf (~> 1.0)
  beaker (~> 3.23)
  beaker-docker (~> 0.2.0)
  beaker-module_install_helper (~> 0.1.5)
  beaker-puppet_install_helper (~> 0.7.1)
  beaker-rspec (~> 6.1)
  docker-api (~> 1.33)
  fast_gettext
  puppet
  puppet-module-posix-default-r2.3
  puppet-module-posix-dev-r2.3
  puppet-module-win-default-r2.3
  puppet-module-win-dev-r2.3
  rbnacl (~> 4.0.2)
  rbnacl-libsodium (~> 1.0)

BUNDLED WITH
   1.15.4
```

**Comment by segiddins** [comment]
I can reproduce, `bundle install` with that Gemfile succeeds on my machine with 1.15.4

**Comment by colby-swandale** [comment]
do you mean you can't reproduce @segiddins?

**Comment by colby-swandale** [comment]
I can't replicate this issue with the same bundler, ruby and rubygem versions

**Comment by segiddins** [comment]
I meant `can't`, sorry

**Comment by colby-swandale** [comment]
ping @bennettp123 is this still a problem?

**Comment by bennettp123** [comment]
@colby-swandale it is on 1.15.4, just trying to figure out how to install 1.16.x. (#5999 suggests it might be fixed in that version).

**Comment by bennettp123** [comment]
`gem install --update bundler --pre`, easier than expected.

It's fixed in bundler-1.16.0.pre.2

**Comment by colby-swandale** [comment]
thank you for testing!
